swagger.yaml

```diff
@@ -1,75 +1,298 @@
-host: localhost
+  description: "This provides an API for interacting with a database of cards from the video game Slay the Spire. It allows users to retrieve cards, add new cards, update cards, and delete cards from the database."
+  version: "1.0"
+  title: "Slay the Spire Cards API"
+
+tags:
+- name: "cards"
+  description: "Contains information on cards"
+
 schemes:
-  - http
-basePath: /api/v0
-consumes:
-  - text/plain
-produces:
-  - text/plain
+- "https"
+
 paths:
-  /:
+  /cards:
+
     get:
-      summary: Get information
-      description: Build and runtime information about the application.
+      tags:
+      - "cards"
+      summary: "Retrieve cards based on search parameters"
+      description: ""
+      operationId: "getCards"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       parameters:
-        - $ref: '#/parameters/pretty'
+        - name: "id"
+          in: "query"
+          description: "id of card in database"
+          required: false
+          type: "integer"
+          format: "int64"
+
+        - name: "types"
+          in: "query"
+          description: "List of types of cards"
+          required: false
+          type: "array"
+          items:
+            type: "string"
+            enum:
+            - "attack"
+            - "skill"
+            - "power"
+            - "status"
+            - "curse"
+            default: "attack,skill,power,status,curse"
+          collectionFormat: "multi"
+
+        - name: "name"
+          in: "query"
+          description: "Card name"
+          required: false
+          type: "string"
+
+        - name: "color"
+          in: "query"
+          description: "List of colors of cards"
+          required: false
+          type: "array"
+          items:
+            type: "string"
+            enum:
+            - "red"
+            - "green"
+            - "blue"
+            - "colorless"
+            default: "red,green,blue,colorless"
+          collectionFormat: "multi"
+
+        - name: "rarity"
+          in: "query"
+          description: "List of rarities of cards"
+          required: false
+          type: "array"
+          items:
+            type: "string"
+            enum:
+            - "basic"
+            - "common"
+            - "uncommon"
+            - "rare"
+            default: "basic,common,uncommon,rare"
+          collectionFormat: "multi"
+
+        - name: "energyMin"
+          in: "query"
+          description: "Minimum value of energy costs of cards (X = 0). Used with energyMax"
+          required: false
+          type: "integer"
+
+        - name: "energyMax"
+          in: "query"
+          description: "Maximum value of energy costs of cards (X = 0). Used with energyMin"
+          required: false
+          type: "integer"
+
+        - name: "keywords"
+          in: "query"
+          description: "List of keywords found in a card's description"
```

Comment:
Just curious, what will this query parameter be searching? I didn't see a description field for the card object.